src/types/webxr.d.ts

```diff
@@ -109,25 +106,30 @@ type XRTargetRayMode = 'gaze'|'tracked-pointer'|'screen';
 interface XRInputSource {
   readonly handedness: XRHandedness;
   readonly targetRayMode: XRTargetRayMode;
+  readonly targetRaySpace: XRSpace;
+  readonly gripSpace?: XRSpace;
+  readonly profiles: Array<String>;
 }
 
 interface XRFrame {
   readonly session: XRSession;
   getViewerPose(referenceSpace?: XRReferenceSpace): XRViewerPose;
-  getInputPose(inputSource: XRInputSource, referenceSpace?: XRReferenceSpace):
-      XRInputPose;
+  getPose(space: XRSpace, referenceSpace: XRReferenceSpace): XRPose;
 }
 
 type XRFrameRequestCallback = (time: number, frame: XRFrame) => void;
 
-interface XRSession extends EventTarget {
+interface XRRenderState {
   baseLayer: XRLayer;
-  requestReferenceSpace(options: XRReferenceSpaceOptions):
-      Promise<XRReferenceSpace>;
-  requestHitTest(
-      origin: Float32Array, direction: Float32Array,
-      frameOfReference: XRFrameOfReference): Promise<XRHitResult[]>;
-  getInputSources(): Array<XRInputSource>;
+}
+
+interface XRSession extends EventTarget {
+  renderState: XRRenderState;
+  updateRenderState(obj: any): any;
```

Comment:
Is this really all the typing we have for this method?

Reply:
I've added more specific type info using XRRenderStateInit.